differ/differ.go

```diff
@@ -301,6 +309,97 @@ func evaluateFilterExpression(eventFilter string, thresholds EventThresholds, ev
+	for _, report := range reports {
+		reportRuleName := string(report.RuleID)[strings.LastIndex(string(report.RuleID), ".")+1:]
+		if reportRuleName == string(ruleName) && report.ErrorKey == errorKey {
+			return report, nil
+		}
+	}
+	return types.RenderedReport{}, errors.New("report for given rule ID nad error key has not been found")
+}
+
+func createServiceLogEntry(report types.RenderedReport, cluster types.ClusterEntry) types.ServiceLogEntry {
+	logEntry := types.ServiceLogEntry{
+		ClusterUUID: string(cluster.ClusterName),
+		Description: report.Description,
+		ServiceName: "CCX Notification Service",
+		Summary:     report.Reason,
+	}
+
+	// It is necessary to truncate the fields because of Service Log limitations
+	if len(logEntry.Description) > serviceLogMaxTextLength {
+		logEntry.Description = logEntry.Description[:serviceLogMaxTextLength]
+	}
+
+	if len(logEntry.Summary) > serviceLogMaxTextLength {
+		logEntry.Summary = logEntry.Summary[:serviceLogMaxTextLength]
+	}
+
+	return logEntry
+}
+
+func produceEntriesToServiceLog(config conf.ConfigStruct, cluster types.ClusterEntry,
+	ruleContent types.RulesMap, reports []types.ReportItem) {
+	renderedReports, err := renderReportsForCluster(
+		conf.GetDependenciesConfiguration(config), cluster.ClusterName,
+		reports, ruleContent)
+	if err != nil {
+		log.Err(err).
+			Str("cluster name", string(cluster.ClusterName)).
+			Msg("Rendering reports failed for this cluster")
+		return
+	}
+	for _, r := range reports {
+		module := r.Module
+		ruleName := moduleToRuleName(module)
+		errorKey := r.ErrorKey
+
+		likelihood, impact, totalRisk, _ := findRuleByNameAndErrorKey(ruleContent, ruleName, errorKey)
+		eventValue := EventValue{
+			Likelihood: likelihood,
+			Impact:     impact,
+			TotalRisk:  totalRisk,
+		}
+
+		// try to evaluate event filter expression
+		result, err := evaluateFilterExpression(serviceLogEventFilter,
+			serviceLogEventThresholds, eventValue)
+
+		if err != nil {
+			log.Err(err).Msg(evaluationErrorMessage)
+			continue
+		}
+
+		if result > 0 {
+			renderedReport, err := findRenderedReport(renderedReports, ruleName, errorKey)
+			if err != nil {
+				log.Err(err).Msgf("Output from content template renderer does not contain "+
+					"result for cluster %s, rule %s and error key %s", cluster.ClusterName, ruleName, errorKey)
+				continue
+			}
+
+			logEntry := createServiceLogEntry(renderedReport, cluster)
+
+			msgBytes, err := json.Marshal(logEntry)
+			if err != nil {
+				log.Error().Err(err).Msg(invalidJSONContent)
+				continue
+			}
+
+			_, _, err = serviceLogNotifier.ProduceMessage(msgBytes)
+			if err != nil {
+				log.Err(err).
+					Str(clusterAttribute, string(cluster.ClusterName)).
+					Str(ruleAttribute, string(ruleName)).
+					Str(errorKeyAttribute, string(errorKey)).
+					Msgf("Sending entry to service log failed for this report")
+				continue
```

Comment:
this "continue" statement at this point looks useless, the loop will go to the next iteration anyways since it's the last instruction of the loop it self (hope it's clear :) )